### PR TITLE
[WFLY-12802] Quickstart for MP Config 1.4

### PR DESCRIPTION
**Pull request by xstefank**
https://issues.redhat.com/browse/WFLY-12802

### CODE REVIEW

**Comment by xstefank** [comment]
@emmartins I believe this is ready for review. Only OpenShift tests are missing which are not a part of this PR.

File: microprofile-config/README.adoc [hunk 1]
```diff
@@ -0,0 +1,793 @@
...
+== Creating the Maven Project
+
+[source,options="nowrap"]
+----
+mvn archetype:generate \
+    -DgroupId=org.wildfly.quickstart.microprofile.config \
+    -DartifactId=microprofile-config \
+    -DinteractiveMode=false \
+    -DarchetypeGroupId=org.apache.maven.archetypes \
+    -DarchetypeArtifactId=maven-archetype-webapp
```
Comment (rhusar) [inline microprofile-config/README.adoc]: Just wondering, why not use the wildfly archetype?

Comment (xstefank) [reply microprofile-config/README.adoc]: @rhusar is there any without Java/Jakarta mentions? I think that is why I chose to stick with default maven ones.

Comment (darranl) [reply microprofile-config/README.adoc]: If we could use any WildFly archetypes we should have merged Stefano's archetype - the problem is none of them are presently available to us once we use the quickstarts with EAP.

File: microprofile-config/src/test/java/org/wildfly/quickstart/microprofile/health/config/MicroProfileConfigIT.java [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Simple tests for MicroProfile Config quickstart. Arquillian deploys an @AR archive to the application server, which
+ * contains several endpoints exposing injected configuration values and verifies that they are correctly invoked.
+ *
+ * @author <a href="mstefank@redhat.com>Martin Stefanko</a>
```
Comment (rhusar) [inline microprofile-config/src/test/java/org/wildfly/quickstart/microprofile/health/config/MicroProfileConfigIT.java]: Missing ending "

File: microprofile-config/src/test/java/org/wildfly/quickstart/microprofile/health/config/MicroProfileConfigIT.java [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+import javax.ws.rs.core.Response;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+/**
+ * Simple tests for MicroProfile Config quickstart. Arquillian deploys an @AR archive to the application server, which
```
Comment (rhusar) [inline microprofile-config/src/test/java/org/wildfly/quickstart/microprofile/health/config/MicroProfileConfigIT.java]: s/@AR/WAR

**Comment by rhusar** [comment]
@xstefank Needs a rebase now

**Pull request merged**